lib/TerminalDisplay.cpp

```diff
@@ -2346,26 +2346,25 @@ void TerminalDisplay::mouseReleaseEvent(QMouseEvent* ev)
 void TerminalDisplay::getCharacterPosition(const QPoint& widgetPoint,int& line,int& column) const
 {
     line = (widgetPoint.y()-contentsRect().top()-_topMargin) / _fontHeight;
+    if (line < 0)
+        line = 0;
+    if (line >= _usedLines)
+        line = _usedLines - 1;
 
     if ( _fixedFont )
         column = (widgetPoint.x() + _fontWidth/2 -contentsRect().left()-_leftMargin) / _fontWidth;
     else
     {
-        int x = contentsRect().left() + widgetPoint.x() - _fontWidth/2;
+        int x = widgetPoint.x() - contentsRect().left() - _fontWidth/2;
```

Comment:
This line needs one more item `- _leftMargin`. By the way, it will be even better to factor out `widgetPoint.x() - contentsRect().left() - _leftMargin` for both fixed- and non-fixed- fonts.